diff --git a/modules/contextual/contextual.module b/modules/contextual/contextual.module
index 0fa148e..651bf20 100644
--- a/modules/contextual/contextual.module
+++ b/modules/contextual/contextual.module
@@ -121,7 +121,6 @@ function views_admin_link($type, $object) {
         'attributes' => array('class' => 'icon-edit'),
         'query' => array('destination' => $_GET['q']),
       );
-      if ($display_id = $object->display_handler->display->id) {
         $links['views-edit']['fragment'] = 'views-tab-'. $display_id;
       }
     }
