diff --git a/sites/all/modules/contextual/contextual.module b/sites/all/modules/contextual/contextual.module
index 0fa148e..2f5f223 100644
--- a/sites/all/modules/contextual/contextual.module
+++ b/sites/all/modules/contextual/contextual.module
@@ -121,7 +121,7 @@ function views_admin_link($type, $object ) {
         'attributes' => array('class' => 'icon-edit'),
         'query' => array('destination' => $_GET['q']),
       );
-      if ($display_id = $object->display_handler->display->id ) {
+      if (isset($object)) && ($display_id = $object->display_handler->display->id ) {
         $links['views-edit']['fragment'] = 'views-tab-'. $display_id;
       
}
     })
}
}
